Update changelog for v0.11 (#3480)
openenclave/openenclave: CHANGELOG.md

[Unreleased][Unreleased_log]
--------------

> [v0.11.0[[v0.11.0_log]

Reviewer: ```suggestion
[v0.11.0][v0.11.0_log]
```

--------------

### Added

> - Open Enclave SDK Release packages can now be built on non-SGX and non-FLC machines.

Reviewer: ```suggestion
- Open Enclave SDK release packages can now be built on non-SGX and non-FLC machines.
```

- Support for arbitrarily large thread-local data for SGX machines.
- Experimental support for OpenSSL inside enclaves has been added while building the SDK from source.
     - Use BUILD_OPENSSL flag while compiling the SDK
     - [OpenSSLSupport.md](docs/OpenSSLSupport.md) documents supported options and configuration needed to OpenSSL inside an enclave.

> 

Reviewer 2: Please add these:

- Custom claims buffer serialization/de-serialization helper functions
- `oe_verify_evidence()` and `oe_verify_attestation_certificate_with_evidence()` added to host-verify package
- SGX attestation endorsement claims from `oe_verify_evidence()`
  - `OE_CLAIM_SGX_TCB_INFO`
  - `OE_CLAIM_SGX_TCB_ISSUER_CHAIN`
  - `OE_CLAIM_SGX_PCK_CRL`
  - `OE_CLAIM_SGX_ROOT_CA_CRL`
  - `OE_CLAIM_SGX_CRL_ISSUER_CHAIN`
  - `OE_CLAIM_SGX_QE_ID_INFO`
  - `OE_CLAIM_SGX_QE_ID_ISSUER_CHAIN`

Author: I changed the wording a little. Can you please verify that it is accurate?

### Changed
- Rename the custom claims buffer added by oe_get_evidence from "custom_claims" to "custom_claims_buffer". Likewise, replace the `OE_CLAIM_CUSTOM_CLAIMS` definition for this name with `OE_CLAIM_CUSTOM_CLAIMS_BUFFER`.
- Building SDK from source		

>       - HAS_QUOTE_PROVIDER cmake option has been removed. This is a continutation of the work in previous release to allo the same build of OE SDK to run on both FLC and non FLC machines."

Reviewer 3: ```suggestion
      - HAS_QUOTE_PROVIDER cmake option has been removed. This is a continuation of the work in previous release to allow the same build of OE SDK to run on both FLC and non-FLC machines."
```

      - Intel SGX EnclaveCommonAPI pacakges are no longer needed to build the SDK.

> 

Reviewer 2: Please add these to "Changed" list:

- `oe_verify_attestation_certificate_with_evidence()` can verify certificates generated by `oe_generate_attestation_certificate()` as well as `oe_get_attestation_certificate_with_evidence()`.
- SGX attestation evidence internal structure changed. Current structure (version 3) is not compatible with previous version. Applications that call `oe_get_evidence()` or `oe_verify_evidence()` have to be rebuilt.
- Some SGX attestation format IDs are renamed:

| Old | New |
| -- | - |
OE_FORMAT_UUID_SGX_ECDSA_P256  | OE_FORMAT_UUID_SGX_ECDSA |
OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT | OE_FORMAT_UUID_LEGACY_REPORT_REMOTE |
OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE | OE_FORMAT_UUID_RAW_SGX_QUOTE_ECDSA |

> 

Reviewer 4: Should sample change be reflected in CHANGELOG as well?

- The attestation functions in local_attestation/remote_attestation/attested_tls/host_verify samples are changed. Now they use attestation plugin APIs, defined in `attestation/attester.h and attestation/verifier.h` to generate and verify evidence.

Author: Done.

### Removed
- COMPILE_SYSTEM_EDL has been removed.

> 

Reviewer 2: Please add this to "Remove" list:

- The following SGX attestation format ID is not yet availble:
  - `OE_FORMAT_UUID_SGX_ECDSA_P384`

[Unreleased][Unreleased_log]
--------------

[v0.11.0][v0.11.0_log]
--------------

### Added
- Open Enclave SDK release packages can now be built on non-SGX and non-FLC machines.
- Support for arbitrarily large thread-local data for SGX machines.
- Experimental support for OpenSSL inside enclaves has been added while building the SDK from source.
     - Use BUILD_OPENSSL flag while compiling the SDK.
     - [OpenSSLSupport.md](docs/OpenSSLSupport.md) documents supported options and configuration needed to use OpenSSL inside an enclave.
- Custom claims buffer serialization/de-serialization helper functions

> oe_verify_evidence() and oe_verify_attestation_certificate_with_evidence() have been added to the host-verify package.

Reviewer 2: This is a bullet by itself:

**-** oe_verify_evidence() and oe_verify_attestation_certificate_with_evidence() have been added to the host-verify package.
